pandas/core/groupby/groupby.py

```diff
@@ -1750,7 +1750,9 @@ def objs_to_bool(vals: ArrayLike) -> tuple[np.ndarray, type]:
             if is_object_dtype(vals.dtype):
                 # GH#37501: don't raise on pd.NA when skipna=True
                 if skipna:
-                    func = np.vectorize(lambda x: bool(x) if not isna(x) else True)
+                    func = np.vectorize(
```

Comment:
is there a reason we don't just do `isna(vals)` here?

Reply:
I think you're suggesting `isna(vals) | np.vectorize(lambda x: bool(x))`. The issue is that `isna(NA)` raises a TypeError.

Reply:
why is 

isna(val's) not enough here?

Reply:
This line is evaluating the truthy-ness of each element, not just determining whether they are null or not. E.g.

    [True, False, pd.NA, None, []]

results in

    array([ True, False,  True,  True, False])

Reply:
Also, `vals.astype(bool, copy=False)` also raises a TypeError when `pd.NA` is an element.